conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/UnclosableOutputStreams.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.exceptions.SafeIoException;
+import io.undertow.io.BufferWritableOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+/** Helper stream used in {@link ConjureBodySerDe} to make error propagation easier. */
+final class UnclosableOutputStreams {
+
+    /**
+     * Wraps the provided {@link OutputStream} such that {@link OutputStream#close()} is a no-op.
+     * This implementation ensures that streams like {@code UndertowOutputStream} are wrapped in a
+     * way that preserves the {@link BufferWritableOutputStream} interface.
+     */
+    static OutputStream wrap(OutputStream delegate) {
+        if (delegate instanceof BufferWritableOutputStream) {
+            return new UnclosableBufferWritableOutputStream(delegate, (BufferWritableOutputStream) delegate);
```

Comment:
no change suggested. Just slightly annoying that we have to do this because `OutputStream` is an abstract class and `BufferWritableOutputStream` is an interface. I initially expected `BufferWritableOutputStream` to be a more specific `OutputStream`, and we could make use of polymorphism to initialize `UnclosableBufferWritableOutputStream`. This is fine though. 🤷